### ISSUE TITLE
Retroarch Switch Core - Arkanoid Returns crash on startup

### ISSUE THREAD

**Issue by melterx12** [finalburnneo/FBNeo#1188]
**Rules and recommendations**
- don't request support for currently unsupported hardware (arcade boards, consoles and computers)
- try to provide as much information as possible (screenshot, savestate, platform, settings, ...)
- make sure it happens with our latest official build from https://github.com/finalburnneo/FBNeo/releases/tag/latest
- any issue on derivated projects that doesn't happen with our latest official build must be reported to that derivated project's maintainers instead

Per title, Arkanoid Returns crashes Retroarch on starting the game. I tested the same rom on Retroarch FBNeo core on my PC as well as the Retroarch Switch FBAlpha core and both of those methods work fine.

Can provide additional info as needed.

**Comment by barbudreadmon** [comment]
>Can provide additional info as needed.

Please provide logs

**Comment by melterx12** [comment]
Title redirection means I am launching retroarch the proper way on the Switch with full RAM access, just in case anyone is wondering.

Nothing is displayed at all, just a black screen before crash. Checked some other Taito F3 games looks like they are all crashing (Tested Land Maker and Puzzle Bobble 4, same issue, very similar log output as below)

Log Below (Debug Verbosity):
`[INFO] RetroArch 1.11.0 (Git 9586d67)
[INFO] === Build =======================================
[INFO] Capabilities: 
[INFO] Built: Sep 30 2022
[INFO] Version: 1.11.0
[INFO] Git: 9586d67
[INFO] =================================================
[INFO] [Input]: Found input driver: "switch".
[INFO] [Environ]: SET_SUBSYSTEM_INFO.
[INFO] Special game type: CBS ColecoVision
  Ident: cv
  ID: 1
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Fairchild ChannelF
  Ident: chf
  ID: 15
  Content:
[INFO]     Rom (required)
[INFO] Special game type: MSX 1
  Ident: msx
  ID: 4
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec PC-Engine
  Ident: pce
  ID: 5
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec SuperGrafX
  Ident: sgx
  ID: 7
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec TurboGrafx-16
  Ident: tg16
  ID: 9
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Nintendo Entertainment System
  Ident: nes
  ID: 11
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Nintendo Family Disk System
  Ident: fds
  ID: 12
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega GameGear
  Ident: gg
  ID: 2
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega Master System
  Ident: sms
  ID: 8
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega Megadrive
  Ident: md
  ID: 3
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega SG-1000
  Ident: sg1k
  ID: 6
  Content:
[INFO]     Rom (required)
[INFO] Special game type: SNK Neo Geo Pocket
  Ident: ngp
  ID: 14
  Content:
[INFO]     Rom (required)
[INFO] Special game type: ZX Spectrum
  Ident: spec
  ID: 10
  Content:
[INFO]     Rom (required)
[INFO] Special game type: Neogeo CD
  Ident: neocd
  ID: 13
  Content:
[INFO]     Iso (required)
[INFO] [Overrides]: Redirecting save file to "/retroarch/cores/savefiles/arkretrn.srm".
[INFO] [Overrides]: Redirecting save state to "/retroarch/cores/savestates/arkretrn.state".
[INFO] [Environ]: SET_SERIALIZATION_QUIRKS.
[INFO] [Environ]: GET_LOG_INTERFACE.
[INFO] [Environ]: GET_MESSAGE_INTERFACE_VERSION.
[INFO] [Environ]: SET_AUDIO_BUFFER_STATUS_CALLBACK.
[INFO] [Content]: Content loading skipped. Implementation will load it on its own.
[INFO] [Environ]: GET_SAVE_DIRECTORY.
[libretro INFO] Setting save dir to /retroarch/cores/savefiles
[INFO] [Environ]: SYSTEM_DIRECTORY: "/retroarch/cores/system".
[libretro INFO] Setting system dir to /retroarch/cores/system
[libretro DEBUG] [FBNeo] nSwitchCode 0x00 : P1 Coin
[libretro DEBUG] [FBNeo] nSwitchCode 0x01 : P1 Start
[libretro DEBUG] [FBNeo] nSwitchCode 0x02 : P1 Up
[libretro DEBUG] [FBNeo] nSwitchCode 0x03 : P1 Down
[libretro DEBUG] [FBNeo] nSwitchCode 0x04 : P1 Left
[libretro DEBUG] [FBNeo] nSwitchCode 0x05 : P1 Right
[libretro DEBUG] [FBNeo] nSwitchCode 0x06 : P1 Button 1
[libretro DEBUG] [FBNeo] nSwitchCode 0x07 : P1 Button 2
[libretro DEBUG] [FBNeo] nSwitchCode 0x08 : P1 Button 3
[libretro DEBUG] [FBNeo] nSwitchCode 0x09 : P1 Button 4
[libretro DEBUG] [FBNeo] nAxisNum 0x01 : P1 Analog Dial
[libretro DEBUG] [FBNeo] nSwitchCode 0x0A : P2 Coin
[libretro DEBUG] [FBNeo] nSwitchCode 0x0B : P2 Start
[libretro DEBUG] [FBNeo] nSwitchCode 0x0C : P2 Up
[libretro DEBUG] [FBNeo] nSwitchCode 0x0D : P2 Down
[libretro DEBUG] [FBNeo] nSwitchCode 0x0E : P2 Left
[libretro DEBUG] [FBNeo] nSwitchCode 0x0F : P2 Right
[libretro DEBUG] [FBNeo] nSwitchCode 0x10 : P2 Button 1
[libretro DEBUG] [FBNeo] nSwitchCode 0x11 : P2 Button 2
[libretro DEBUG] [FBNeo] nSwitchCode 0x12 : P2 Button 3
[libretro DEBUG] [FBNeo] nSwitchCode 0x13 : P2 Button 4
[libretro DEBUG] [FBNeo] nAxisNum 0x02 : P2 Analog Dial
[libretro DEBUG] [FBNeo] nSwitchCode 0x14 : P3 Coin
[libretro DEBUG] [FBNeo] nSwitchCode 0x15 : P3 Start
[libretro DEBUG] [FBNeo] nSwitchCode 0x16 : P3 Up
[libretro DEBUG] [FBNeo] nSwitchCode 0x17 : P3 Down
[libretro DEBUG] [FBNeo] nSwitchCode 0x18 : P3 Left
[libretro DEBUG] [FBNeo] nSwitchCode 0x19 : P3 Right
[libretro DEBUG] [FBNeo] nSwitchCode 0x1A : P3 Button 1
[libretro DEBUG] [FBNeo] nSwitchCode 0x1B : P3 Button 2
[libretro DEBUG] [FBNeo] nSwitchCode 0x1C : P3 Button 3
[libretro DEBUG] [FBNeo] nSwitchCode 0x1D : P3 Button 4
[libretro DEBUG] [FBNeo] nSwitchCode 0x1E : P4 Coin
[libretro DEBUG] [FBNeo] nSwitchCode 0x1F : P4 Start
[libretro DEBUG] [FBNeo] nSwitchCode 0x20 : P4 Up
[libretro DEBUG] [FBNeo] nSwitchCode 0x21 : P4 Down
[libretro DEBUG] [FBNeo] nSwitchCode 0x22 : P4 Left
[libretro DEBUG] [FBNeo] nSwitchCode 0x23 : P4 Right
[libretro DEBUG] [FBNeo] nSwitchCode 0x24 : P4 Button 1
[libretro DEBUG] [FBNeo] nSwitchCode 0x25 : P4 Button 2
[libretro DEBUG] [FBNeo] nSwitchCode 0x26 : P4 Button 3
[libretro DEBUG] [FBNeo] nSwitchCode 0x27 : P4 Button 4
[INFO] [Environ]: SET_CONTROLLER_INFO.
[INFO]    Controller port: 1
[INFO]       Classic (ID: 5)
[INFO]       Modern (ID: 517)
[INFO]       Mouse (ball only) (ID: 773)
[INFO]       Mouse (full) (ID: 514)
[INFO]       Pointer (ID: 6)
[INFO]       Lightgun (ID: 4)
[INFO]    Controller port: 2
[INFO]       Classic (ID: 5)
[INFO]       Modern (ID: 517)
[INFO]       Mouse (ball only) (ID: 773)
[INFO]       Mouse (full) (ID: 514)
[INFO]       Pointer (ID: 6)
[INFO]       Lightgun (ID: 4)
[libretro INFO] Initialize DIP switches.
[INFO] [Environ]: GET_CORE_OPTIONS_VERSION.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_V2.
[INFO] [Environ]: GET_VARIABLE: fbneo-cpu-speed-adjust = "100%"
[INFO] [Environ]: GET_VARIABLE: fbneo-allow-depth-32 = "enabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-vertical-mode = "disabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-force-60hz = "disabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-frameskip-type = "disabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-frameskip-manual-threshold = "33"
[INFO] [Environ]: GET_VARIABLE: fbneo-fixed-frameskip = "0"
[INFO] [Environ]: GET_VARIABLE: fbneo-diagnostic-input = "Hold Start"
[INFO] [Environ]: GET_VARIABLE: fbneo-hiscores = "enabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-allow-patched-romsets = "disabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-samplerate = "48000"
[INFO] [Environ]: GET_VARIABLE: fbneo-sample-interpolation = "4-point 3rd order"
[INFO] [Environ]: GET_VARIABLE: fbneo-fm-interpolation = "4-point 3rd order"
[INFO] [Environ]: GET_VARIABLE: fbneo-lowpass-filter = "enabled"
[INFO] [Environ]: GET_VARIABLE: fbneo-analog-speed = "100%"
[INFO] [Environ]: GET_VARIABLE: fbneo-lightgun-crosshair-emulation = "hide with lightgun device"
[libretro INFO] [FBNeo] Searching all possible locations for romset arkretrn
[libretro INFO] [FBNeo] Romset found at /roms/arkretrn
[libretro INFO] [FBNeo] No romset found at /retroarch/cores/system/fbneo/arkretrn
[libretro INFO] [FBNeo] No romset found at /retroarch/cores/system/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0xb50cfb92 and name e36-11.20 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0xc940dba1 and name e36-10.19 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0xf16985e0 and name e36-09.18 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x4467ff37 and name e36-15.17 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x1ea8558b and name e36-03.12 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x694eda31 and name e36-02.8 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x54b9b2cd and name e36-01.4 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x266bf1c1 and name e36-07.47 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x110ab729 and name e36-06.45 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0xdb18bce2 and name e36-05.43 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x3bae39be and name e36-12.32 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x94448e82 and name e36-13.33 from archive /roms/arkretrn
[libretro INFO] [FBNeo] Using ROM with known crc 0x2250959b and name e36-04.38 from archive /roms/arkretrn
[libretro INFO] [FBNeo] No missing files, proceeding
[libretro INFO] [FBNeo] Samplerate set to 48000
[INFO] [Environ]: GET_VARIABLE: fbneo-dipswitch-arkretrn-Music_Tempo_(must_restart!) = "Normal / Fast"
[DEBUG] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[libretro INFO] [FBNeo] Applied dipswitches from core options
[libretro DEBUG] Cheat cpu-register INIT.
[libretro DEBUG] CPU-registry: 68k cpu #0 ...
[libretro DEBUG] CPU-registry: 68k cpu #1 ...`

**Comment by barbudreadmon** [comment]
From where it's crashing, i can only think there is some problem with those 2 lines on switch : 

https://github.com/finalburnneo/FBNeo/blob/35ec4af081e816cbf3e8c362a3fe870f73bd0a8a/src/burn/drv/taito/d_taitof3.cpp#L1304-L1305

For now i commented the second one as a test fix on the libretro port (it's for debugging and not required for emulation, but maybe there is something illegal about it on switch), let me know if the issue disappears after updating your core, then we'll try to figure out the exact reason for the crash.

**Comment by melterx12** [comment]
yep that fixed it - all the games are working now thanks!👍 

should I close this issue ?

**Comment by barbudreadmon** [comment]
Nope, i'd like to figure out what is wrong exactly with this line so i'll ask you for further testing soon if you don't mind.

**Comment by barbudreadmon** [comment]
I pushed another test fix, let me know the results after it's available from the buildbot (maybe backup your current core for safety)

**Comment by melterx12** [comment]
is it still building? tried updating cores just now through the online updater but it says no update was found

**Comment by barbudreadmon** [comment]
There is some time lag before new builds become available from the online updater, try again in 1-2 hours

**Comment by melterx12** [comment]
got the updated core, all the games are still working 👍

**Comment by barbudreadmon** [comment]
Indeed, that would be exactly the same problem, the switch port is somehow crashing on "%S" format specifiers, even though i'm replacing them by "%s" format specifiers at runtime.
I might have an idea about what's going on, looks like the libretro guys made a backward compatibility breaking change without warning.

**Comment by barbudreadmon** [comment]
i pushed a fix, i expect it will fix the problem everywhere

**Comment by melterx12** [comment]
the Mushihime games and tatio games are all working now! Thanks!

**Comment by barbudreadmon** [comment]
I'll close then, thanks for the report !
